plugins/modules/domain_record.py

```diff
@@ -191,14 +191,15 @@ def _find_record(
         self, domain: Domain, name: str, rtype: str, target: str
     ) -> Optional[DomainRecord]:
         # We should strip the FQDN from the user-defined record name if defined.
-        suffix = "." + domain.domain
-        search_name = name
-        if search_name.endswith(suffix):
-            search_name = search_name[: -len(suffix)]
+        name = name.removesuffix("." + domain.domain)
```

Comment:
We can use `.removesuffix(...)` here now that Python 3.9 is our minimum supported version.